weave

```diff
@@ -783,6 +789,13 @@ dns_args() {
             -h|--hostname|--hostname=*)
                 HOSTNAME_SPECIFIED=1
                 ;;
+            --add-host)
+                DNS_EXTRA_HOSTS_ARGS="--add-host=$2 $DNS_EXTRA_HOSTS_ARGS"
+                shift
+                ;;
+            --add-host=*)
+                DNS_EXTRA_HOSTS_ARGS="--add-host=${1#*=} $DNS_EXTRA_HOSTS_ARGS"
+                ;;
```

Comment:
What's the rationale for doing this in here rather than having a separate function?

As things stand, we will generally invoke `dns_args` twice, which is very expensive since we do all the work twice, including a call to weave (to obtain the domain).